src/mach/exports.rs

```diff
@@ -274,10 +274,16 @@ impl<'a> ExportTrie<'a> {
     /// Create a new, lazy, zero-copy export trie from the `DyldInfo` `command`
     pub fn new(bytes: &'a [u8], command: &load_command::DyldInfoCommand) -> Self {
         let start = command.export_off as usize;
-        let end = (command.export_size + command.export_off) as usize;
+        let end = start.saturating_add(command.export_size as usize);
```

Comment:
Since this silently truncates maybe should do checked add and warn when it’s too large ? I assume this would only ever happen on a 32-bit system reading the 6GB binary ?

Reply:
I added some `log::warn` for this and the `FatArch` case.
This here I found via fuzzing, though the `FatArch` was indeed a valid file.